server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -168,16 +176,24 @@ public int getNumOpenConnections()
       final QueryMetrics<? super Query<T>> queryMetrics = toolChest.makeMetrics(query);
       queryMetrics.server(host);
 
+      long timeoutAt = ((Long) context.get(QUERY_FAIL_TIME)).longValue();
```

Comment:
Rationale for me is:

- Queries making other queries is going outside the normal framework. The normal framework is a query comes in to the broker, then fans out to historicals/other data nodes, then fans back in and the broker does a merge and returns results. There's not a standard place for "make another subquery" to fit in.
- Queries making other queries may lead to metrics not being completely collected, or security rules not being properly applied (some of which are checked in the resources), or requests not being properly logged, or exceptions not being properly alerted on. This needs to be thought through.
- Users are still free to write extensions where queries make other queries, but due to the above two points, I don't think that should be considered a "public api" at this time.